File: PO/Project/JavaProject/src/Animals/Fox.java
package Animals;

import Abstractions.*;
import com.google.gson.JsonObject;

import java.awt.*;

public class Fox extends Animal {

    public Fox(World world, Point position) {
        super(4, 4, position, world);
        this.avatar = 'F';
        this.image_path = "Fox.png";
        loadImage();
    }

    @Override
    public int fight(Organism target) {
        if (target.getStrength() > this.strength) {
            world.getLogger().log("Fox avoided the fight");

            world.findField(this.position).setMember(this);

            return 1;
        }

        return super.fight(target);
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Fox(world, newPosition);
    }
}

File: Sem2/PO/Project/C++/src/fox.cpp
#include "fox.h"

fox::fox(world* world_point, YX position)
:animal(4,4, position, world_point)
{
    avatar = 'F';
}

int fox::Fight(organizm* target)
{
    if (target->GetStrength() > strength)
    {
        world_point->GetLogger()->Log("fox avoided the fight");

        world_point->FindField(position)->member = this;

        return 1;
    }
    
    return animal::Fight(target);
}

organizm* fox::MakeChild(YX new_position)
{
    return (new fox(world_point, new_position));
}

fox::~fox()
{
}
